lib/ansible/plugins/lookup/fileglob.py

```diff
@@ -76,7 +77,8 @@ def run(self, terms, variables=None, **kwargs):
             for dwimmed_path in found_paths:
                 if dwimmed_path:
                     globbed = glob.glob(to_bytes(os.path.join(dwimmed_path, term_file), errors='surrogate_or_strict'))
-                    ret.extend(to_text(g, errors='surrogate_or_strict') for g in globbed if os.path.isfile(g))
-                    if ret:
+                    term_results.extend(to_text(g, errors='surrogate_or_strict') for g in globbed if os.path.isfile(g))
+                    if term_results:
```

Comment:
I'm not 100% sure this is correct, or at least maybe not exact enough, although it probably works correct.

You could probably just do:
```
term_results = [to_text(g, errors='surrogate_or_strict') for g in globbed if os.path.isfile(g)]
if term_results:
    ret.extend(term_results)
    break
```

Reply:
Effectively, it's the same result. But your code is slightly more efficient with less lines, so I changed it.